test/integration/client-side-encryption/client_side_encryption.prose.test.js

```diff
@@ -1424,4 +1424,245 @@ describe('Client Side Encryption Prose Tests', metadata, function () {
+              failCommands: ['aggregate']
+            }
+          });
+      });
+
+      it('expects an error and a command failed event', async function () {
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect an exception to be thrown from the network error. Expect a CommandFailedEvent.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.message).to.include('closed');
+        }
+        expect(result).to.not.exist;
+        expect(aggregateFailed.failure.message).to.include('closed');
+        expect(aggregateStarted).to.exist;
+      });
+    });
+
+    context('Case 3: Decrypt Error', function () {
+      it('errors on decryption but command succeeds', async function () {
+        // Use ``encryptedClient`` to insert the document ``{ "encrypted": <malformedCiphertext> }``
+        // into ``db.decryption_events``.
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect an exception to be thrown from the decryption error.
+        // Expect a CommandSucceededEvent. Expect the CommandSucceededEvent.reply
+        // to contain BSON binary for the field
+        // ``cursor.firstBatch.encrypted``.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        await collection.drop();
+        await collection.insertOne({ encrypted: malformedCiphertext });
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error.message).to.include('HMAC validation failure');
+        }
+        expect(result).to.not.exist;
+        const doc = aggregateSucceeded.reply.cursor.firstBatch[0];
+        expect(doc.encrypted).to.be.instanceOf(Binary);
+        expect(aggregateStarted).to.exist;
+      });
+    });
+
+    context('Case 4: Decrypt Success', function () {
+      it('succeeds on decryption and command succeeds', async function () {
+        // Use ``encryptedClient`` to insert the document ``{ "encrypted": <ciphertext> }``
+        // into ``db.decryption_events``.
+        // Use ``encryptedClient`` to run an aggregate on ``db.decryption_events``.
+        // Expect no exception.
+        // Expect a CommandSucceededEvent. Expect the CommandSucceededEvent.reply
+        // to contain BSON binary for the field ``cursor.firstBatch.encrypted``.
+        const collection = encryptedClient.db('').collection('decryption_events');
+        await collection.drop();
+        await collection.insertOne({ encrypted: cipherText });
+        let result;
+        try {
+          result = await collection.aggregate([]).toArray();
+        } catch (error) {
+          expect(error).to.not.exist;
+        }
+        expect(result[0].encrypted).to.equal('hello');
```

Comment:
I actually assert the value of the field here is the original value decrypted, and then in line 1663 is the important part showing the same field in the command succeeded event is still an encrypted `Binary`